lib/Fhp/Action/GetSepaDirectDebitLeadTime.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Fhp\Action;
+
+use Fhp\BaseAction;
+use Fhp\Segment\DME\HIDMESv1;
+use Fhp\Segment\DME\HIDMESv2;
+use Fhp\Segment\DME\MinimaleVorlaufzeitSEPALastschrift;
+use Fhp\Segment\DSE\HIDSESv1;
+
+/**
+ * Retrieves information about lead times for SEPA Direct Debit Requests
+ */
+class GetSepaDirectDebitLeadTime extends BaseAction
+{
+    const SEQUENCE_TYPES = ['FRST', 'OOFF', 'FNAL', 'RCUR'];
+    const CORE_TYPES = ['CORE', 'COR1'];
+
+    /** @var MinimaleVorlaufzeitSEPALastschrift|null */
+    private $leadTime;
+
+    /** @var string */
+    private $coreType;
+
+    /** @var string */
+    private $seqType;
+
+    /** @var bool */
+    private $singleDirectDebit;
+
+    public static function create(string $seqType, bool $singleDirectDebit, string $coreType = 'CORE')
+    {
+        if (!in_array($coreType, self::CORE_TYPES)) {
+            throw new \InvalidArgumentException('Unknown CORE Type');
+        }
+        if (!in_array($seqType, self::SEQUENCE_TYPES)) {
+            throw new \InvalidArgumentException('Unknown SEPA Sequence Type');
+        }
+        $result = new GetSepaDirectDebitLeadTime();
+        $result->coreType = $coreType;
+        $result->seqType = $seqType;
+        $result->singleDirectDebit = $singleDirectDebit;
+
+        return $result;
+    }
+
+    /** {@inheritdoc} */
+    public function createRequest($bpd, $upd)
+    {
+        $type = $this->singleDirectDebit ? 'HIDSES' : 'HIDMES';
+
+        $hidxes = $bpd->getLatestSupportedParameters($type);
+
```

Comment:
Hier fehlt ein null-Check. Oder oben `requireLatestSupportedParameters()` verwenden.

Reply:
Ich finde die Fehlermeldungen von requireLatestSupportedParameters verbesserungsdürftig. Kann man da evtl. die verständlichere Fehlermeldung als zweiten Parameter übergeben?

Reply:
Kann man machen, denke ich.

Reply:
Auch wichtig ist hier, ob `array_key_exists($type, $this->parameters)`. Wenn ja (aber `getLatestSupportedParameters($type)===null`), dann unterstützt die Bank zwar den Geschäftsvorfall, aber in keiner Version, die phpFinTS auch unterstützt. Man müsste sich also an die phpFinTS-Entwickler wenden, um die richtige Version nachzurüsten. Wenn `!array_key_exists`, dann unterstützt die Bank den Geschäftsvorfall überhaupt nicht, also muss man sich entweder an die Bank wenden oder nach Alternativen suchen.

Und bei `supportsParameters()` wäre noch erwähnenswert, dass der `@return`-Wert unabhängig davon ist, ob die entsprechenden Segmente in der Bibliothek auch unterstützt werden, anders als bei `getLatestSupportedParameters()`.